packages/engine/src/datastore/batch/flush.rs

```diff
@@ -2,36 +2,74 @@ use crate::datastore::{
     arrow::{meta_conversion::get_dynamic_meta_flatbuffers, padding},
     error::Result,
     prelude::*,
+    storage::BufferChange,
 };
 
-pub trait GrowableArrayData: Sized + std::fmt::Debug {
-    fn _len(&self) -> usize;
-    fn _null_count(&self) -> usize;
-    fn _null_buffer(&self) -> Option<&[u8]>;
-    fn _get_buffer(&self, index: usize) -> &[u8];
-    fn _get_non_null_buffer_count(&self) -> usize;
-    fn _child_data(&self) -> &[Self];
+/// The info required about Arrow array data in order to grow it
+///
+/// Can be implemented by ArrayData, ArrayDataRef, Python FFI array
+/// data.
+pub(in crate::datastore) trait GrowableArrayData:
+    Sized + std::fmt::Debug
+{
+    fn len(&self) -> usize;
+    fn null_count(&self) -> usize;
+    fn null_buffer(&self) -> Option<&[u8]>;
+    fn buffer(&self, index: usize) -> &[u8];
+    /// Arrow stores the null buffer separately from other buffers.
+    fn num_buffers_except_null_buffer(&self) -> usize;
```

Comment:
I also thought about it, but it sounds like it might be the number of non-null buffers (i.e. number of valid buffers, whose pointer isn't null), rather than the number of buffers that aren't the null buffer. What about `num_buffers_except_null_bitmap`?

Reply:
It's better but I don't really like `except` in this context. @Alfred-Mountfield any thoughts?